plugins/modules/openssl_signature.py

```diff
@@ -0,0 +1,383 @@
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: openssl_signature
+short_description: Sign and verify data with openssl
+description:
+    - This module allows one to sign and verify data via a certificate and a private key
+    - The module can use the cryptography Python library, or the pyOpenSSL Python
+      library. By default, it tries to detect which one is available. This can be
+      overridden with the I(select_crypto_backend) option. Please note that the PyOpenSSL backend
+      was deprecated in Ansible 2.9 and will be removed in Ansible 2.13.
+requirements:
+    - Either cryptography >= 1.4 (some key types require newer versions)
+    - Or pyOpenSSL >= 0.11 (Ed25519 and Ed448 keys are not supported with this backend)
+author:
+    - Patrick Pichler (@aveexy)
+    - Markus Teufelberger (@MarkusTeufelberger)
+options:
+    action:
+        description: Action to be executed
+        type: str
+        required: true
+        choices: [ sign, verify ]
+    privatekey_path:
+        description:
+            - The path to the private key to use when signing
+        type: path
+    privatekey_passphrase:
+        description:
+            - The passphrase for the private key.
+            - This is required if the private key is password protected.
+        type: str
+    path:
+        description: File to sign/verify
+        type: path
+        required: true
+    certificate:
+        description: Certificate required for verify action
+        type: path
+    signature:
+        description: Base64 encoded signature required for verify action
+        type: str
+    select_crypto_backend:
+        description:
+            - Determines which crypto backend to use.
+            - The default choice is C(auto), which tries to use C(cryptography) if available, and falls back to C(pyopenssl).
+            - If set to C(pyopenssl), will try to use the L(pyOpenSSL,https://pypi.org/project/pyOpenSSL/) library.
+            - If set to C(cryptography), will try to use the L(cryptography,https://cryptography.io/) library.
+        type: str
+        default: auto
+        choices: [ auto, cryptography, pyopenssl ]
+notes:
+    - |
+      When using the C(cryptography) backend, the following key types require at least the following C(cryptography) version:
+      RSA keys: C(cryptography) >= 1.4
+      DSA and ECDSA keys: C(cryptography) >= 1.5
+      ed448 and ed25519 keys: C(cryptography) >= 2.6
+seealso:
+    - module: x509_certificate
+    - module: openssl_privatekey
+'''
+
+EXAMPLES = r'''
+- name: Sign example file
+  openssl_signature:
+    action: sign
+    privatekey_path: private.key
+    path: /tmp/example_file
+  register: sig
+
+- name: Verify signature of example file
+  openssl_signature:
+    action: verify
+    certificate: cert.pem
+    path: /tmp/example_file
+    signature: "{{ sig.signature }}"
+'''
+
+RETURN = r'''
+signature:
+    description: Base64 encoded signature
+    returned: changed or success
```

Comment:
```suggestion
    returned: success
```
Changed is part of success :)